### DISCUSSION TITLE
Is it possible to limit the number of displayed lines in a code block?

### DISCUSSION THREAD

**Discussion by hellt** [facelessuser/pymdown-extensions#2247] [Q&A] [answered]
If the code block is large, is it possible to make it scrollable?

**Comment by facelessuser** [comment, marked as answer]
Yep, just add a special class and some CSS to trigger off the class turns it into a scrollable code block.

````
``` {.python .code-scroll}
some long code...
```
````

**Reply by hellt** [reply]
The `code-scroll` class should have some custom css defined for it?
If yes I'd appreciate someone sharing it here. I hoped it is a common use case...

**Reply by facelessuser** [reply]
Something like this should work. Just select whatever height you want.

```css
    div.highlight.code-scroll pre > code {
      max-height: 300px;
    }
```

I used to do scrolling code for large blocks until I started using expandable code previews, but that's a lot more complicated to get started with.

<img width="742" alt="Screenshot 2023-11-23 at 11 42 28 AM" src="https://github.com/facelessuser/pymdown-extensions/assets/1055125/ab791b72-0356-42b5-8a55-e75303c3400b">

**Reply by hellt** [reply]
ohhh, I would LOVE to get expandable code previews!
I once logged this in https://github.com/squidfunk/mkdocs-material/issues/2171 but despite it getting some votes from others Martin didn't find it essential.

I wonder, would it be possible to **fund** this custom work to make it available?

**Reply by hellt** [reply]
thanks for your help @facelessuser 
I now see this had nothing to do with the awesome plugins you/team authored.

Nevertheless, this helped me achieve the goal https://github.com/srl-labs/learn-srlinux/pull/103/commits/8b79447572b99b86789de47dba566cabff5d9372 

I opted in using `vh` vertical height, will see if that is going to cost me in the future.

**Reply by facelessuser** [reply]
Well, I've provided details for anyone to do this here: https://github.com/squidfunk/mkdocs-material/issues/4964#issuecomment-1718085465

**Reply by hellt** [reply]
Thanks @facelessuser, I will check there if anyone wants to be contracted/sponsored to make this plugin easily integrate-able.
